tests/test_coef.py

```diff
@@ -0,0 +1,722 @@
+    assert parts is not None
+    assert len(parts) == 0
+
+    parts = _get_parts(feature0, (20,))
+    assert parts is not None
+    assert len(parts) == 0
+
+
+def test_cdist_parts_one_vs_one():
+    from scipy.spatial.distance import cdist
+    from sklearn.metrics import adjusted_rand_score as ari
+
+    parts0 = np.array(
+        [
+            [1, 1, 2, 2, 3, 3],
+        ]
+    )
+    parts1 = np.array(
+        [
+            [3, 3, 1, 1, 2, 2],
+        ]
+    )
+
+    expected_cdist = cdist(parts0, parts1, metric=ari)
+    np.testing.assert_array_equal(expected_cdist, np.array([[1.0]]))
+
+    observed_cdist = cdist_parts(parts0, parts1)
+
+    np.testing.assert_array_equal(observed_cdist, expected_cdist)
+
+
+def test_cdist_parts_one_vs_one_dissimilar():
+    from scipy.spatial.distance import cdist
+    from sklearn.metrics import adjusted_rand_score as ari
+
+    parts0 = np.array(
+        [
+            [1, 1, 2, 1, 3, 3],
+        ]
+    )
+    parts1 = np.array(
+        [
+            [3, 3, 1, 1, 2, 3],
+        ]
+    )
+
+    expected_cdist = cdist(parts0, parts1, metric=ari)
+    np.testing.assert_array_equal(expected_cdist, np.array([[-0.022727272727272728]]))
+
+    observed_cdist = cdist_parts(parts0, parts1)
+
+    np.testing.assert_array_equal(observed_cdist, expected_cdist)
+
+
+def test_cdist_parts_one_vs_two():
+    from scipy.spatial.distance import cdist
+    from sklearn.metrics import adjusted_rand_score as ari
+
+    parts0 = np.array(
+        [
+            [1, 1, 2, 1, 3, 3],
+        ]
+    )
+    parts1 = np.array(
+        [
+            [3, 3, 1, 1, 2, 3],
+            [3, 3, 1, 1, 2, 2],
+        ]
+    )
+
+    expected_cdist = cdist(parts0, parts1, metric=ari)
+    np.testing.assert_array_equal(
+        expected_cdist,
+        np.array(
+            [
+                [-0.022727272727272728, 0.4444444444444444],
+            ]
+        ),
+    )
+
+    observed_cdist = cdist_parts(parts0, parts1)
+
+    np.testing.assert_array_equal(observed_cdist, expected_cdist)
+
+
+def test_cdist_parts_two_vs_two():
+    from scipy.spatial.distance import cdist
+    from sklearn.metrics import adjusted_rand_score as ari
+
+    parts0 = np.array(
+        [
+            [1, 1, 2, 2, 3, 3],
+            [1, 1, 2, 1, 3, 3],
+        ]
+    )
+    parts1 = np.array(
+        [
+            [3, 3, 1, 1, 2, 3],
+            [3, 3, 1, 1, 2, 2],
+        ]
+    )
+
+    expected_cdist = cdist(parts0, parts1, metric=ari)
+    np.testing.assert_array_equal(
+        expected_cdist,
+        np.array(
+            [
+                [0.4444444444444444, 1.0],
+                [-0.022727272727272728, 0.4444444444444444],
+            ]
+        ),
+    )
+
+    observed_cdist = cdist_parts(parts0, parts1)
+
+    np.testing.assert_array_equal(observed_cdist, expected_cdist)
+
+
+def test_get_coords_from_index():
+    # data is an example with n_obj = 5, not used in fact
+    data = np.array(
```

Comment:
Yes, you're right, it is not needed. I commented out that code. I wanted to let the example there for illustration purposes since otherwise I got lost after taking a look at the code again :joy: